gsy_framework/sim_results/carbon_emissions.py

```diff
@@ -0,0 +1,169 @@
+from typing import Union, Dict, List
+
+import requests
+import pandas as pd
+from pendulum import duration
+from entsoe import EntsoePandasClient
+from entsoe.mappings import lookup_area, Area
+from entsoe.parsers import parse_generation
+
+from gsy_framework.sim_results.results_abc import ResultsBaseClass
+
+URL = "https://web-api.tp.entsoe.eu/api"
+
+GENERATION_PLANT_TO_CARBON_EMISSIONS = {
+    # source: https://www.ipcc.ch/site/assets/uploads/2018/02/ipcc_wg3_ar5_annex-iii.pdf#page=7
```

Comment:
Can you please explain why you hardcoded this here and not call the API? Is this because of your comment in line 27?

Reply:
Yes, it follows the adaption for the code in https://github.com/EnergieID/entsoe-py. Once we update to python 3.11 all of these extra methods will be removed.